Sources/MapboxMaps/Foundation/Extensions/NilEncoder.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+internal struct NilEncoder {
+    var userInfo: [CodingUserInfoKey: Any]
```

Comment:
Can we provide some constructor to simplify `. shouldEncodeNilValues` key passing? Right now I see no reason why this struct called `NilEncoder` and not something like `ExtraUserInfoEncoder`.

Reply:
good point, I fixed it now.